components/calendar/calendar.js

```diff
@@ -0,0 +1,474 @@
+			'en': { show: 'Show {month}' },
+			'es': { show: 'Show {month} ' },
+			'fr': { show: 'Show {month} ' },
+			'ja': { show: 'Show {month} ' },
+			'ko': { show: 'Show {month} ' },
+			'nl': { show: 'Show {month} ' },
+			'pt': { show: 'Show {month} ' },
+			'sv': { show: 'Show {month} ' },
+			'tr': { show: 'Show {month} ' },
+			'zh': { show: 'Show {month} ' },
+			'zh-tw': { show: 'Show {month} ' }
+		};
+	}
+
+	constructor() {
+		super();
+
+		const descriptor = getDateTimeDescriptor();
+		this._monthNames = descriptor.calendar.months.long;
+		this._firstDayOfWeek = descriptor.calendar.firstDayOfWeek;
+		const weekdays = descriptor.calendar.days.long;
+		const weekdaysShort = descriptor.calendar.days.short;
+
+		this._arrangedWeekdays = [];
+		for (let i = this._firstDayOfWeek; i < this._firstDayOfWeek + 7; i++) {
+			const index = i % 7;
+			const day = {
+				long: weekdays[index],
+				short: weekdaysShort[index]
+			};
+			this._arrangedWeekdays.push(day);
+		}
+
+		this._dialogLabelId = getUniqueId();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.addEventListener('keydown', this._onKeyDown);
+		this.addEventListener('d2l-calendar-selected', (e) => {
+			this._selectedDate = e.detail.date;
+		});
+
+		let selected;
+		if (this.selectedValue) {
+			selected = parseDate(this.selectedValue);
+		} else {
+			selected = new Date();
+		}
+
+		this._selectedDate = {
+			month: selected.getMonth(),
+			date: selected.getDate(),
+			year: selected.getFullYear()
+		};
+
+		this._shownMonth = this._selectedDate.month;
+		this._shownYear = this._selectedDate.year;
+
+		let focusDay = 1;
+		if (this._selectedDate.month === this._shownMonth) focusDay = this._selectedDate.date;
+		this._focusDate = {
+			date: focusDay,
+			month: this._shownMonth,
+			year: this._shownYear
+		};
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldVal, prop) => {
+			if (prop === '_selectedDate') {
+				this._generateDaysInMonth();
+			}
+			if (prop === '_shownMonth') {
+				this._onMonthChange();
+				this._generateDaysInMonth();
+				this.updateComplete.then(() => {
+					this.updateComplete.then(() => {
+						this._getNodeAndAddFocus(this._keyboardTriggeredMonthChange);
+						this._keyboardTriggeredMonthChange = false;
+					});
+				});
+			}
+		});
+	}
+
+	render() {
+		const weekdayHeaders = this._arrangedWeekdays.map((day) => html`
+			<th role="columnheader" abbr="${day.long}">
+				<abbr title="${day.long}">
+					${day.short}
+				</abbr>
+			</th>
+		`);
+
+		let dayRows;
+		if (this._dates) {
+			dayRows = this._dates.map((week) => {
+				const weekHtml = week.map((day) => {
+					return html`
+						<td>
+							<d2l-calendar-date
+								date=${day.date}
+								month=${day.month}
+								year=${day.year}
+								?focused=${day.focused}
+								?other-month=${day.otherMonth}
+								?selected=${day.selected}>
+							</d2l-calendar-date>
```

Comment:
It was because this component was getting large, but I have moved things back into one component now and I think it has actually ended up simplifying things.